backend/dataall/modules/shares_base/services/share_item_service.py

```diff
@@ -0,0 +1,334 @@
+                max_length=50,
+                lowercase=True,
+                regex_pattern='[^a-zA-Z0-9-]',
+                separator='-',
+            )
+            log.info(f'S3AccessPointName={s3_access_point_name}')
+
+            if not share_item:
+                share_item = ShareObjectItem(
+                    shareUri=uri,
+                    itemUri=item_uri,
+                    itemType=item_type,
+                    itemName=item.name,
+                    status=ShareItemStatus.PendingApproval.value,
+                    owner=context.username,
+                )
+                session.add(share_item)
+        return share_item
+
+    @staticmethod
+    @ResourcePolicyService.has_resource_permission(REMOVE_ITEM, parent_resource=_get_share_uri)
+    def remove_shared_item(uri: str):
+        with get_context().db_engine.scoped_session() as session:
+            share_item = ShareObjectRepository.get_share_item_by_uri(session, uri)
+            if (
+                share_item.itemType == ShareableType.Table.value
+                and share_item.status == ShareItemStatus.Share_Failed.value
+            ):
+                share = ShareObjectRepository.get_share_by_uri(session, share_item.shareUri)
+                ResourcePolicyService.delete_resource_policy(
+                    session=session,
+                    group=share.groupUri,
+                    resource_uri=share_item.itemUri,
+                )
+
+            item_sm = ShareItemSM(share_item.status)
+            item_sm.run_transition(ShareItemActions.RemoveItem.value)
+            ShareObjectRepository.remove_share_object_item(session, share_item)
+        return True
+
+    @staticmethod
+    @ResourcePolicyService.has_resource_permission(GET_SHARE_OBJECT)
+    def resolve_shared_item(uri, item: ShareObjectItem):
+        with get_context().db_engine.scoped_session() as session:
+            return ShareObjectRepository.get_share_item(session, item.itemType, item.itemUri)
+
+    @staticmethod
+    def check_existing_shared_items(share):
+        with get_context().db_engine.scoped_session() as session:
+            return ShareObjectRepository.check_existing_shared_items(session, share.shareUri)
+
+    @staticmethod
+    def list_shareable_objects(share, filter, is_revokable=False):
+        states = None
+        if is_revokable:
+            states = ShareItemSM.get_share_item_revokable_states()
+
+        with get_context().db_engine.scoped_session() as session:
+            return ShareObjectRepository.list_shareable_items(session, share, states, filter)
+
+    @staticmethod
+    @ResourcePolicyService.has_resource_permission(LIST_ENVIRONMENT_SHARED_WITH_OBJECTS)
+    def paginated_shared_with_environment_datasets(session, uri, data) -> dict:
+        return ShareObjectRepository.paginate_shared_datasets(session, uri, data)
+
+    @staticmethod
+    def _get_glue_database_for_share(glueDatabase, account_id, region):
```

Comment:
nit: this is a "private" method that I don't think is being used anywhere

Reply:
I'll handle it in part8 where I have to refactor the share_item_service